_docs-sources/pipelines/overview/configurations-as-code.md

```diff
@@ -0,0 +1,390 @@
+
+A typical approach to building Pipelines configurations is to first define minimal configurations that address the most common use-cases, and then to refactor and generalize those configurations as needed to reduce repetition.
+
+More details on how these configurations are defined will be detailed below.
+
+## Primer on HCL Terminology
+
+HCL is an extensible configuration language developed by HashiCorp. See [this](https://github.com/hashicorp/hcl/blob/main/hclsyntax/spec.md) for the full specification.
+
+The main terminology you need to know to understand the documentation below includes:
+
+- **Blocks**: A block is a collection of nested configurations that are defined within curly braces `{}`.
+
+  :::tip
+  A `filter` block is nested in the `environment` block in the example above.
+  :::
+
+- **Attributes**: An attribute is a key-value pair separated by an `=` that is defined within a block.
+
+  :::tip
+  The `paths` attribute is defined within the `filter` block in the example above.
+  :::
+
+- **Labels**: A label is one or more strings that are used to qualify a block.
+
+  :::tip
+  The `an_environment` label is used to qualify the `environment` block in the example above.
+  :::
+
+  Note that some blocks have multiple labels. Labels can both serve as a mechanism for disambiguating blocks from each other, and as a way to have concrete implementations of general behavior.
+
+  :::tip
+  In the `authentication_profile` block, the first label of `aws_oidc` is used to qualify the block as an `aws_oidc` authentication profile block, which means that the content inside the block is relevant for `aws_oidc` authentication profiles, and the second label of `profile` is used to qualify the block as an `aws_oidc` authentication profile block named `profile`.
+  :::
+
+## Global Configurations
+
+Any configurations located within a `.gruntwork` directory either in the current working directory, or a parent directory of the current working directory are referred to as global configurations. These configurations are typically applicable within a wide range of contexts within a repository, and are the primary mechanism for configuring Pipelines.
+
+Pipelines will attempt to find exactly one directory named `.gruntwork` when it is attempting to discover configurations. It will not continue to search for configurations in parent directories once it finds a `.gruntwork` directory.
+
+Note that you will frequently see filenames for configurations within the `.gruntwork` directory that are named after the configuration block that they define. This is a common pattern, but not a requirement. Any `.hcl` file found within a `.gruntwork` directory will be parsed by Pipelines as one global configuration.
```

Comment:
There isn't a hierarchy of configurations, there can only be (up to) one `.gruntwork` directory read by the configuration parser and only (up to) one `gruntwork.hcl` file that will be read by the configuration parser.

The reason we don't require that the `.gruntwork` directory exists at the root of the git repository is that:

1. This supports arbitrary folder structure. You might have all your Terragrunt code in a `terragrunt` directory, in which case it would be nice to have the `.gruntwork` directory nested in there so that your paths are all relative to your `terragrunt` directory. 
2. It works outside of git. This is highly convenient for `pipelines` tests, as we don't have to create an extra `.git` directory in order to have the `.gruntwork` directory detected within a `tests` folder. It also makes it so that we don't have to use special logic for handling recursive traversal outside of `.terragrunt-cache` directories, as cloned OpenTofu modules have their own git directories, and we have to make sure that Pipelines doesn't accidentally think of that directory as the root of configuration.